### ISSUE TITLE
clang 6 warnings

### ISSUE THREAD

**Issue by tjhei** [GeodynamicWorldBuilder/WorldBuilder#123]
I am getting the following warnings with 5dbad3a
```
ninja
[303/329] Building CXX object CMakeFiles/aspect.dir/contrib/WorldBuilder/source/features/interface.cc.o
../contrib/WorldBuilder/source/features/interface.cc:165:37: warning: implicit conversion turns floating-point number into integer: 'double' to 'const int' [-Wfloat-conversion]
                  const int parts = std::ceil(length / maximum_distance_between_coordinates);
                            ~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
[328/329] Building CXX object CMakeFiles/aspect.dir/contrib/WorldBuilder/source/features/fault.cc.o
../contrib/WorldBuilder/source/features/fault.cc:340:48: warning: implicit conversion turns floating-point number into integer: 'const __gnu_cxx::__alloc_traits<std::allocator<double> >::value_type' (aka 'const double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/fault.cc:340:76: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/fault.cc:342:48: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_segment = distance_from_planes["segment"];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/fault.cc:468:48: warning: implicit conversion turns floating-point number into integer: 'const __gnu_cxx::__alloc_traits<std::allocator<double> >::value_type' (aka 'const double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/fault.cc:468:76: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/fault.cc:470:48: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_segment = distance_from_planes["segment"];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
6 warnings generated.
[328/329] Building CXX object CMakeFiles/aspect.dir/contrib/WorldBuilder/source/features/subducting_plate.cc.o
../contrib/WorldBuilder/source/features/subducting_plate.cc:351:48: warning: implicit conversion turns floating-point number into integer: 'const __gnu_cxx::__alloc_traits<std::allocator<double> >::value_type' (aka 'const double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/subducting_plate.cc:351:76: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/subducting_plate.cc:353:48: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_segment = distance_from_planes["segment"];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/subducting_plate.cc:476:48: warning: implicit conversion turns floating-point number into integer: 'const __gnu_cxx::__alloc_traits<std::allocator<double> >::value_type' (aka 'const double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/subducting_plate.cc:476:76: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const unsigned int current_section = one_dimensional_coordinates[distance_from_planes["section"]];
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/features/subducting_plate.cc:478:48: warning: implicit conversion turns floating-point number into integer: 'std::map<std::__cxx11::basic_string<char>, double, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, double> > >::mapped_type' (aka 'double') to 'const unsigned int' [-Wfloat-conversion]
          const unsigned int current_segment = distance_from_planes["segment"];
                             ~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
6 warnings generated.
[328/329] Building CXX object CMakeFiles/aspect.dir/contrib/WorldBuilder/source/utilities.cc.o
../contrib/WorldBuilder/source/utilities.cc:515:64: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const double reference_on_side_of_line = (point_list[next_section][0] - point_list[current_section][0])
                                                    ~~~~~~~~~~ ^~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:515:94: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const double reference_on_side_of_line = (point_list[next_section][0] - point_list[current_section][0])
                                                                                  ~~~~~~~~~~ ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:516:87: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                   * (reference_point[1] - point_list[current_section][1])
                                                                           ~~~~~~~~~~ ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:517:66: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                   - (point_list[next_section][1] - point_list[current_section][1])
                                                      ~~~~~~~~~~ ^~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:517:96: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                   - (point_list[next_section][1] - point_list[current_section][1])
                                                                                    ~~~~~~~~~~ ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:518:87: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                   * (reference_point[0] - point_list[current_section][0])
                                                                           ~~~~~~~~~~ ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:523:40: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const Point<2> P1(point_list[current_section]);
                            ~~~~~~~~~~ ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:525:40: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
          const Point<2> P2(point_list[next_section]);
                            ~~~~~~~~~~ ^~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:596:62: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  total_average_angle = plane_segment_angles[original_current_section][0][0]
                                        ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:597:85: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                        + fraction_CPL_P1P2 * (plane_segment_angles[original_next_section][0][0]
                                                               ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:598:87: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                               - plane_segment_angles[original_current_section][0][0]);
                                                                 ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:692:82: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
              for (unsigned int i_segment = 0; i_segment < plane_segment_lengths[original_current_section].size(); i_segment++)
                                                           ~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:749:49: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  WBAssert(plane_segment_angles[original_next_section].size() > current_segment,
                           ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/include/world_builder/assert.h:30:14: note: expanded from macro 'WBAssert'
      if (! (condition)) { \
             ^~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:751:89: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                           << ", and current_segment.size() = " << plane_segment_angles[original_next_section].size());
                                                                   ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/include/world_builder/assert.h:33:58: note: expanded from macro 'WBAssert'
                   << " at line " << __LINE__ << ": " << message << std::endl; \
                                                         ^~~~~~~
../contrib/WorldBuilder/source/utilities.cc:762:81: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  const double interpolated_angle_top    = plane_segment_angles[original_current_section][current_segment][0]
                                                           ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:762:107: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  const double interpolated_angle_top    = plane_segment_angles[original_current_section][current_segment][0]
                                                           ~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:763:104: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_angles[original_next_section][current_segment][0]
                                                                                  ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:763:127: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_angles[original_next_section][current_segment][0]
                                                                                  ~~~~~~~~~~~~~~~~~~~~                        ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:764:106: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_angles[original_current_section][current_segment][0])
                                                                                    ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:764:132: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_angles[original_current_section][current_segment][0])
                                                                                    ~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:767:81: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  const double interpolated_angle_bottom = plane_segment_angles[original_current_section][current_segment][1]
                                                           ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:767:107: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  const double interpolated_angle_bottom = plane_segment_angles[original_current_section][current_segment][1]
                                                           ~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:768:104: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_angles[original_next_section][current_segment][1]
                                                                                  ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:768:127: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_angles[original_next_section][current_segment][1]
                                                                                  ~~~~~~~~~~~~~~~~~~~~                        ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:769:106: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_angles[original_current_section][current_segment][1])
                                                                                    ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:769:132: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_angles[original_current_section][current_segment][1])
                                                                                    ~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:773:82: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  double interpolated_segment_length     = plane_segment_lengths[original_current_section][current_segment]
                                                           ~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:773:108: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                  double interpolated_segment_length     = plane_segment_lengths[original_current_section][current_segment]
                                                           ~~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:774:105: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_lengths[original_next_section][current_segment]
                                                                                  ~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:774:128: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                           + fraction_CPL_P1P2 * (plane_segment_lengths[original_next_section][current_segment]
                                                                                  ~~~~~~~~~~~~~~~~~~~~~                        ^~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:775:107: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_lengths[original_current_section][current_segment]);
                                                                                    ~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
../contrib/WorldBuilder/source/utilities.cc:775:133: warning: implicit conversion turns floating-point number into integer: 'const double' to 'std::vector::size_type' (aka 'unsigned long') [-Wfloat-conversion]
                                                                                  - plane_segment_lengths[original_current_section][current_segment]);
                                                                                    ~~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~~~~~~
32 warnings generated.

```

**Comment by MFraters** [comment]
hmm, all implicit conversion warnings. Should be easy to fix.

**Comment by MFraters** [comment]
@tjhei could you test whether the fix gets rid of all the warnings?

**Comment by MFraters** [comment]
fixed by #124
